taichi/codegen/llvm/codegen_llvm.cpp

```diff
@@ -2515,6 +2528,16 @@ void TaskCodeGenLLVM::visit(MeshPatchIndexStmt *stmt) {
   llvm_val[stmt] = get_arg(2);
 }
 
+void TaskCodeGenLLVM::visit(MatrixInitStmt *stmt) {
+  auto type = tlctx->get_data_type(stmt->ret_type->as<TensorType>());
+  llvm::Value *vec = llvm::UndefValue::get(type);
```

Comment:
Looks like the only reason you need this UndefValue is to pass the type info to `builder->CreateInsertElement()`

To make the code more explicit, can we try switching to this overloaded `builder->CreateInsertElement()` interface, where you directly pass-in the `Type` instead of having to wrap it with some UndefValue?

![2022-08-17 11-52-13 的屏幕截图](https://user-images.githubusercontent.com/22334008/185031312-6ff07d72-41a2-4d8a-98a9-25dcd7c78fc1.png)